packages/lib/isOutOfBounds.tsx

```diff
@@ -17,50 +21,253 @@ function guardAgainstBookingInThePast(date: Date) {
   throw new BookingDateInPastError();
 }
 
-function isOutOfBounds(
-  time: dayjs.ConfigType,
-  {
-    periodType,
-    periodDays,
-    periodCountCalendarDays,
-    periodStartDate,
-    periodEndDate,
-  }: Pick<
-    EventType,
-    "periodType" | "periodDays" | "periodCountCalendarDays" | "periodStartDate" | "periodEndDate"
-  >,
-  minimumBookingNotice?: number
-) {
+/**
+ * Dates passed to this function are timezone neutral.
+ */
+export function calculatePeriodLimits({
```

Comment:
Refactored this logic out of `isOutOfBounds` so that it isn't recomputed for every timeslot.